lib/Compiler.js

```diff
@@ -148,6 +167,8 @@ class Compiler extends Tapable {
 				stats.startTime = startTime;
 				stats.endTime = Date.now();
 				this.hooks.done.callAsync(stats, err => {
+					this.running = false;
```

Comment:
Every `err` in `return callback(err);` should also set running to `false`.

Best create a `finalCallback` to handle this without much code duplication:

``` js
const finalCallback = (err, stats) => {
  this.running = false;
  callback(err, stats);
}
```